src/Auth0.php

```diff
@@ -522,24 +546,25 @@ public function exchange()
             throw new ApiException('Invalid access_token - Retry login.');
         }
 
-        $accessToken = $response['access_token'];
+        $this->setAccessToken($response['access_token']);
 
-        $refreshToken = false;
         if (isset($response['refresh_token'])) {
-            $refreshToken = $response['refresh_token'];
+            $this->setRefreshToken($response['refresh_token']);
         }
 
-        $idToken = false;
-        if (isset($response['id_token'])) {
-            $idToken = $response['id_token'];
+        if (! empty($response['id_token'])) {
+            $this->setIdToken($response['id_token']);
         }
 
-        $this->setAccessToken($accessToken);
-        $this->setIdToken($idToken);
-        $this->setRefreshToken($refreshToken);
+        if ($this->skipUserinfo) {
+            $user = $this->idTokenDecoded;
+        } else {
+            $user = $this->authentication->userinfo($this->accessToken);
+        }
 
-        $user = $this->authentication->userinfo($accessToken);
-        $this->setUser($user);
+        if ($user) {
```

Comment:
Method requires an array so we need to make sure we have something before attempting to set.